lib/components/base/Avatar.vue

```diff
@@ -34,45 +34,32 @@
   </svg>
 </template>
 
-<script setup>
+<script setup lang="ts">
 import { ref } from 'vue'
 
 const pixelated = ref(false)
-const img = ref(null)
+const img = ref<HTMLImageElement | null>(null)
 
-defineProps({
-  src: {
-    type: String,
-    default: null,
-  },
-  alt: {
-    type: String,
-    default: '',
-  },
-  size: {
-    type: String,
-    default: 'sm',
-    validator(value) {
-      return ['xxs', 'xs', 'sm', 'md', 'lg', 'none'].includes(value)
-    },
-  },
-  circle: {
-    type: Boolean,
-    default: false,
-  },
-  noShadow: {
-    type: Boolean,
-    default: false,
-  },
-  loading: {
-    type: String,
-    default: 'lazy',
-  },
-  raised: {
-    type: Boolean,
-    default: false,
-  },
-})
+withDefaults(
+  defineProps<{
+    src?: string
+    alt: string
+    size: 'xxs' | 'xs' | 'sm' | 'md' | 'lg' | 'none'
+    circle: boolean
+    noShadow: boolean
+    loading: 'lazy' | 'eager'
+    raised: boolean
+  }>(),
+  {
+    src: undefined,
```

Comment:
No need for this, it would be undefined anyway by default.

Reply:
did decide to keep it based on the discussion on discord